test/document_stores/test_memory.py

```diff
@@ -73,6 +74,19 @@ def test_update_bm25(self, documents):
         assert isinstance(bm25_representation, BM25)
         assert bm25_representation.corpus_size == ds.get_document_count()
 
+    @pytest.mark.integration
+    def test_update_bm25_table(self):
+        table_doc = Document(
+            content=pd.DataFrame(columns=["id", "text"], data=[["1", "This is a test"], ["2", "This is another test"]]),
```

Comment:
Just to enhance this test I'd add a non string value. Something simple like an int, or something weirder like an object, simply to see what the behavior is. For example, I think `to_csv` can't deal with objects, so we might want to check for that early, or at least raise a meaningful exception.

Reply:
I changed `"1"` to `1` to have a non-string value in the table. Additionally, just to be sure, I added a call `astype(str)` on the dataframe before calling `to_csv` to be sure it contains only string values.